src/main/scala/io/iohk/ethereum/consensus/ethash/validators/EthashBlockHeaderValidator.scala

```diff
@@ -37,22 +37,21 @@ class EthashBlockHeaderValidator(blockchainConfig: BlockchainConfig) extends Blo
     import scala.collection.JavaConverters._
 
     def getPowCacheData(epoch: Long): PowCacheData = {
-      if (epoch == 0) EthashBlockHeaderValidator.epoch0PowCache else
-        Option(powCaches.get(epoch)) match {
-          case Some(pcd) => pcd
-          case None =>
-            val data = new PowCacheData(
-              cache = EthashUtils.makeCache(epoch),
-              dagSize = EthashUtils.dagSize(epoch))
-
-            val keys = powCaches.keySet().asScala
-            val keysToRemove = keys.toSeq.sorted.take(keys.size - MaxPowCaches + 1)
-            keysToRemove.foreach(powCaches.remove)
-
-            powCaches.put(epoch, data)
-
-            data
-        }
+      Option(powCaches.get(epoch)) match {
```

Comment:
are you sure that this `powCaches.get(epoch)` wouldn't return null in some point? if so maybe consider using 
```
Try(powCaches.get(epoch)).toOption
```
so we would omit NullPointerException

Reply:
I think, it is ok. `Option(null) == None`